src/Sarif/Core/WebResponse.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public partial class WebResponse
+    {
+        [JsonIgnore]
+        public string HttpVersion { get; private set; }
+
+        public static WebResponse Parse(string responseString)
+        {
+            var webResponse = new WebResponse();
+
+            ParseStatusLine(
```

Comment:
>ParseStatusLine [](start = 12, length = 15)

why not parse the string directly into the webResponse object and return?

Reply:
(+ similar pattern in webRequest class)

---
In reply to: [295566231](https://github.com/microsoft/sarif-sdk/pull/1537#discussion_r295566231) [](ancestors = 295566231)

Reply:
You are right. Code is *so* much cleaner that way! Great comment!

---
In reply to: [295566746](https://github.com/microsoft/sarif-sdk/pull/1537#discussion_r295566746) [](ancestors = 295566746,295566231)

Reply:
Saved 45 lines of code.

---
In reply to: [295568125](https://github.com/microsoft/sarif-sdk/pull/1537#discussion_r295568125) [](ancestors = 295568125,295566746,295566231)